src/main/java/bio/terra/stairway/Flight.java

```diff
@@ -287,10 +287,21 @@ private StepResult stepWithRetry() throws InterruptedException, StairwayExecutio
               && context().isDoing()
               && context().getDebugInfo().getFailAtSteps().containsKey(context().getStepIndex())
               && !debugStepsFailed.contains(context().getStepIndex())) {
-            result =
-                new StepResult(
-                    this.context().getDebugInfo().getFailAtSteps().get(context().getStepIndex()));
+            StepStatus failStatus =
```

Comment:
This clause is getting big and is already pretty nested. How about putting in a separate method. Something like:
```
StepResult debugStatusReplacement(FlightContext context, StepResult result)
```
And then if there is no debug info, we return the input result. If there is, we substitute the replacement.

What do you think?

Reply:
SGTM. I went ahead and started calling it on the undo case, even though both of our current status replacements are only for the DO direction. That ok too, or should I leave it as only being called on the DO result (w/ or w/out the then unnecessary context().isDoing() checking)?

Reply:
We do not want to inject failures in the undo, because that just induces DISMAL FAILURE results.

Reply:
Done, thanks.